pkg/controllers/node/termination/terminator/terminator.go

```diff
@@ -79,32 +80,14 @@ func (t *Terminator) Taint(ctx context.Context, node *v1.Node) error {
 // Drain evicts pods from the node and returns true when all pods are evicted
 // https://kubernetes.io/docs/concepts/architecture/nodes/#graceful-node-shutdown
 func (t *Terminator) Drain(ctx context.Context, node *v1.Node) error {
-	// Get evictable pods
-	pods := &v1.PodList{}
-	if err := t.kubeClient.List(ctx, pods, client.MatchingFields{"spec.nodeName": node.Name}); err != nil {
+	pods, err := nodeutil.GetEvictablePods(ctx, t.kubeClient, t.clock.Now(), node)
```

Comment:
It would be nice if we could put the !Active() call inside of the GetEvictablePods(), but I realize that IsEvictable() is wired through here, which needs to have the list of active pods to know when the node is fully drained. Any thoughts on how we can simplify here, as we're still stuck in the same situation (albeit better) as before, where we have distinct pod filtering logic that requires a non-shallow understanding of how everything wires together to know how if it's right.

Reply:
> that requires a non-shallow understanding of how everything wires together to know how if it's right

It's a good callout. The best thing that I can think about is moving the pod length check up and then filtering out the non-active pods before calling evict.

Reply:
Agreed. I'm not sure there's a better way than throwing a comment in there to explain that it's for efficiency (since it really is just to avoid redundant eviction calls).

Reply:
I tried creating two separate functions `IsWaitingEviction()` and `IsEvictable()` but I thought that this added more confusion than the current implementation. Happy to discuss more if you have any more thoughts here.